### ISSUE TITLE
CDN fails with network error, if you get the URL wrong the first time

### ISSUE THREAD

**Issue by benasher44** [CocoaPods/CocoaPods#9103]
🌈
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

I upgraded from CocoaPods 1.7.2 to 1.7.5. I wanted to get us using the new CDN before the 1.8.0 release to suss out any issues.

## What did you do?

I made the following update to our Podfile:
```diff
-source 'https://github.com/CocoaPods/Specs.git'
+source 'https://cdn.cocoapods.org/'
```

## What did you expect to happen?

Something magical where it used the CDN for the first time ⭐️.

## What happened instead?

I got the following message:

```
[!] CDN: cocoapods- Relative path couldn't be downloaded: CocoaPods-version.yml, error: Failed to open TCP connection to cdn.cocoapods.orgCocoaPods-version.yml:443 (getaddrinfo: nodename nor servname provided, or not known)
```

### Stack

```
   CocoaPods : 1.6.1
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
    RubyGems : 3.0.3
        Host : Mac OS X 10.14.6 (18G87)
       Xcode : 10.3 (10G8)
         Git : git version 2.20.1 (Apple Git-117)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : autodesk-bimcocoapods-specs - git@git.autodesk.com:BIM360/BIMCocoaPods-Specs.git @ 6620f26b39184f8997a8e4aecf6d3b010d9b6dde
               cocoapods- -  @ fatal: not a git repository (or any of the parent directories): .git
               master - https://github.com/CocoaPods/Specs.git @ efd4d290b1f702fdfb22d249ef0eb0e0c5b04191
               plangrid - git@github.com:plangrid/ios-specs.git @ f21f31e4b52670b73d0f74820690511b2120c58e
               plangrid-1 - https://github.com/plangrid/ios-specs.git @ fae671e0aa52e53efcfbdbd5339cad38ff1534cb
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# rubocop:disable Metrics/LineLength

require_relative './scripts/verify_netrc'

source '<internal specs repo>'
source 'https://cdn.cocoapods.org/'

workspace 'PlanGrid.xcworkspace'
platform :ios, '11.0'
supports_swift_versions '>= 5.0'

use_frameworks!
inhibit_all_warnings!

ENV['COCOAPODS_DISABLE_STATS'] = 'true'

pod 'AMPopTip', '~> 4.0'
pod 'Bugsnag', '~> 5.17'
pod 'CocoaLumberjack/Swift', '~> 3.4'
pod 'GCDWebServer', '~> 3.0'
pod 'GRDB.swift', '~> 4.0'
pod 'lightstep', '~> 3.2'
pod 'LUUID', '2.0.0'
pod 'PDKeychainBindingsController', git: 'https://github.com/carlbrown/PDKeychainBindingsController.git', commit: '61fff54e2f8dafcfcbab7cc3f3f367a183e2e6a6'
pod 'PGFoundation', '0.4.0.dev.316'
pod 'ReactiveCocoa', '~> 10.0'
pod 'ReactiveObjC', '~> 3.1'
pod 'ReactiveSwift', '~> 6.0'
pod 'sqlite3', '~> 3.25'
pod 'SlackTextViewController', '1.9.6'
pod 'SSZipArchive', '~> 2.1'
pod 'TTTAttributedLabel', '2.0.0'

# Our Pods
pod 'ReactiveLists', '0.5.1'
pod 'PUG', '1.2.1'

# Development dependencies
pod 'FLEX', '~> 3.0', configurations: ['Debug']
pod 'SwiftLint', '0.31.0'

def testing_pods
  supports_swift_versions '>= 4.2'
  pod 'Quick', '~> 2.0'
  pod 'Nimble', '~> 8.0'
end

target 'PlanLib' do
  project 'PlanLib/PlanLib.xcodeproj'
  target 'PlanLibTests' do
    testing_pods
  end
end

target 'PlanUI' do
  project 'PlanUI/PlanUI.xcodeproj'
  target 'PlanUITests' do
    testing_pods
  end
end

target 'PlanGrid' do
  project 'PlanGrid/PlanGrid.xcodeproj'

  pod 'Amplitude-iOS', '~> 4.6'
  pod 'Analytics', '~> 3.0'
  pod 'AppsFlyerFramework', '~> 4.10'
  pod 'Apptimize', '~> 3.1'
  pod 'BSImagePicker', '~> 2.9'
  pod 'FBSDKCoreKit/Core', '~> 5.0'
  pod 'GVRSDK', '~> 1.70'
  pod 'libPhoneNumber-iOS', '0.9.13'

  verify_netrc
  pod 'ModelViewer', '2.4.1'

  pod 'NewRelicAgent', '~> 6.7'
  pod 'PSPDFKit', podspec: 'https://customers.pspdfkit.com/cocoapods/<secret>/8.3.1.podspec'
  pod 'Segment-Amplitude', '~> 2.0'
  pod 'PGBIMLib', '2.2.0'

  target 'PlanGridUnitTests' do
    testing_pods
  end

  target 'PlanGridIntegrationTests' do
    testing_pods
  end
end

target 'PGSharedTestUtils' do
  project 'PGSharedTestUtils/PGSharedTestUtils.xcodeproj'
  testing_pods
end

# Patch GRDB Target to depend on our sqlite3
def patch_grdb_target(target, all_targets)
  # Replace system #includes with our @imports
  system "sed -i '' -e 's/#include <sqlite3.h>/@import sqlite3;/g' '#{File.dirname(__FILE__)}/Pods/GRDB.swift/Support/grdb_config.h'"

  # Ensure sqlite3 is built before GRDB
  target.add_dependency(all_targets.find { |t| t.name == 'sqlite3' })
  target.build_configurations.each do |config|
    # Set the flag to skip importing system sqlite
    config.build_settings['OTHER_SWIFT_FLAGS'] ||= ['$(inherited)', '-D', 'GRDBCUSTOMSQLITE', '-D', 'SQLITE_ENABLE_FTS5']

    # Add sqlite3 to framework search paths
    config.build_settings['FRAMEWORK_SEARCH_PATHS'] ||= ['$(inherited)', '"${PODS_CONFIGURATION_BUILD_DIR}/sqlite3"']
  end
end

def patch_sqlite3_target(target)
  target.build_configurations.each do |config|
    # Set the custom sqlite build flags we want
    config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', 'SQLITE_MAX_VARIABLE_NUMBER=2147483000', 'SQLITE_ENABLE_FTS5=1', 'SQLITE_ENABLE_JSON1=1', 'SQLITE_USE_URI=1']

    # Make Tsan skip known sqlite false positives
    config.build_settings['OTHER_CFLAGS'] ||= ['$(inherited)', '"-fsanitize-blacklist=$(PODS_ROOT)/../TsanBlacklist.txt"']

    # Set framework version to for PGFoundation linking (matches compat version in otool -L /usr/lib/libsqlite3.dylib)
    config.build_settings['DYLIB_CURRENT_VERSION'] = '9.0.0'
    config.build_settings['DYLIB_COMPATIBILITY_VERSION'] = '9.0.0'
  end
end

post_install do |installer|
  # REMOVES SQLITE3 FROM PODS DEPENDENCIES
  Dir.glob("#{File.dirname(__FILE__)}/Pods/Target\\ Support\\ Files/*/*.xcconfig").each do |f|
    system "sed -i '' -e 's/-l\\\"sqlite3\\\" //g' '#{f}'"
    system "sed -i '' -e 's/-l\\\"sqlite3\\\.0\\\" //g' '#{f}'"
  end

  all_targets = installer.pods_project.targets
  all_targets.each do |target|
    patch_sqlite3_target(target) if target.name == 'sqlite3'
    patch_grdb_target(target, all_targets) if target.name == 'GRDB.swift'
  end
end
# rubocop:enable Metrics/LineLength
```

## Project that demonstrates the issue

Internal project, sorry…

**Comment by benasher44** [comment]
Okay I figured out a repro:

1. Add the cocoapods cdn source without the trailing slash like so: `source 'https://cdn.cocoapods.rg`
2. `pod install`, and you'll get the following error:
```
[!] Unable to add a source with url `https://cdn.cocoapods.org` named `cocoapods-`.
You can try adding it manually in `/Users/basher/.cocoapods/repos` or via `pod repo add`.
```
3. Be confused and try fixing the url to have the trailing slash.
4. `pod install`, and you'll get the above error in the original report.

Your cdn source will be broken until you remove it from `~/.cocoapods/repos` and `pod install` again (fixes it).

**Comment by dnkoutso** [comment]
Fixed already in 1.8.0.beta.1, if a typo is made it will add the `/` with 1e2ff6550fee2ef77ded5fff68824cf59414b75c

Let me know if there is anything additional to do and I can re-open.

**Comment by benasher44** [comment]
Ah great! Thanks!
